### ISSUE TITLE
Strange Results when running AARTFAAC Data- Possible sourcefinder bug

### ISSUE THREAD

**Issue by ycendes** [transientskp/banana#89]
I ran the first 24 hour data run on struis through the TraP (struis:/scratch/fhuizing/aartfaac/results/24h), which resulted in strange results as you can see in the following:
![screen shot 2015-07-14 at 12 17 54 pm](https://cloud.githubusercontent.com/assets/4549934/8678072/cbbe9cde-2a52-11e5-99f5-20e37b5a24d3.png)

On closer examination, it appears that the first error in the logs appears with this message

2015-07-13 17:27:10.843 ERROR tkp.sourcefinder.image: Background is masked: cannot fit

This error then continues for each image afterwards, but what's particularly interesting is the oscillations appear to happen in a pattern that follows the errors:

2015-07-13 19:06:39.961 ERROR tkp.sourcefinder.image: Background is masked: cannot fit
2015-07-13 19:06:39.961 ERROR tkp.sourcefinder.image: Background is masked: cannot fit
2015-07-13 19:06:39.961 ERROR tkp.sourcefinder.image: Background is masked: cannot fit
2015-07-13 19:06:39.975 ERROR tkp.sourcefinder.image: All data is masked: cannot fit
2015-07-13 19:06:39.975 ERROR tkp.sourcefinder.image: Background is masked: cannot fit
2015-07-13 19:06:39.994 ERROR tkp.sourcefinder.image: Gaussian fit failed at 303.555949, 698.708657
2015-07-13 19:06:40.006 ERROR tkp.sourcefinder.image: Gaussian fit failed at 303.423369, 699.141403
2015-07-13 19:06:40.006 ERROR tkp.sourcefinder.image: All data is masked: cannot fit
2015-07-13 19:06:40.016 ERROR tkp.sourcefinder.image: Background is masked: cannot fit
2015-07-13 19:06:40.025 ERROR tkp.sourcefinder.image: Background is masked: cannot fit
2015-07-13 19:06:41.343 ERROR tkp.sourcefinder.image: All data is masked: cannot fit
2015-07-13 19:06:41.343 ERROR tkp.sourcefinder.image: All data is masked: cannot fit
2015-07-13 19:06:41.344 ERROR tkp.sourcefinder.image: Background is masked: cannot fit
2015-07-13 19:06:41.344 ERROR tkp.sourcefinder.image: Background is masked: cannot fit
2015-07-13 19:06:41.344 ERROR tkp.sourcefinder.image: Background is masked: cannot fit
2015-07-13 19:06:41.344 ERROR tkp.sourcefinder.image: All data is masked: cannot fit
2015-07-13 19:06:41.344 ERROR tkp.sourcefinder.image: Background is masked: cannot fit
2015-07-13 19:06:41.345 ERROR tkp.sourcefinder.image: Background is masked: cannot fit
2015-07-13 19:06:41.345 ERROR tkp.sourcefinder.image: All data is masked: cannot fit
2015-07-13 19:06:41.345 ERROR tkp.sourcefinder.image: Background is masked: cannot fit
2015-07-13 19:06:41.345 ERROR tkp.sourcefinder.image: Background is masked: cannot fit
2015-07-13 19:06:41.346 ERROR tkp.sourcefinder.image: Background is masked: cannot fit
2015-07-13 19:06:41.357 ERROR tkp.sourcefinder.image: All data is masked: cannot fit
2015-07-13 19:06:41.357 ERROR tkp.sourcefinder.image: Background is masked: cannot fit
2015-07-13 19:06:41.377 ERROR tkp.sourcefinder.image: Gaussian fit failed at 303.547834, 698.720846
2015-07-13 19:06:41.389 ERROR tkp.sourcefinder.image: Gaussian fit failed at 303.415290, 699.153600

I'm guessing that means the oscillation is due to the initial error.

It should be noted that the images themselves when you load them appear normal- the peak image is 21224, which has 1,290 sources.  Curiously though, when Gijs and I first visited its page the image was filled with sources (ie here): http://banana.transientskp.org/r2/vlo_ycendes/image/21224/

But now when I load it or the images around it I get the message that there are "no sources" in the image.  Not sure if this is related to the errors above, or if it's something else?

**Comment by AntoniaR** [comment]
So this is a TraP issue and not Banana....

I'm fairly certain that your sources are not associating correctly (This image claims to have 1354 sources but they must all be on top of each other as you certainly can't see them all on the plot... http://banana.transientskp.org/master/vlo_ycendes/image/58572/ ). Therefore, as each each new image is added, the number of sources are systematically increasing. I saw a less extreme example with my MWA dataset (see issue https://github.com/transientskp/tkp/issues/421).

The oscillation makes me think that the TraP thinks it is dealing with different fields - and hence not doing the sky region correctly.

A major issue for me is the RA values - they are all negative! So again, concerns about the WCS being interpreted correctly.

So I think this boils down to TraP not understanding the AARTFAAC co-ordinate system and is potentially related to the issue that Tim raised: https://github.com/transientskp/tkp/issues/444
Basically, if TraP does not understand the pixel to WCS conversion correctly then sources will not associate and it won't necessarily know which sky region is being considered.

Finally, AARTFAAC images may be another case where the source association needs >1 times bmaj for the source association routine (see issue https://github.com/transientskp/tkp/issues/421). But I think the WCS-pixel is the first issue to check and solve.

**Comment by ycendes** [comment]
> So this is a TraP issue and not Banana....

Mea culpa.  Is there a way to move the issue to the correct place?

The sources were in fact pretty much on top of each other when we examined the image and it still had sources in it, a whole bunch of them in the middle it looked like.  So you may well be right about the coordinates.  How can we check?

**Comment by gijzelaerr** [comment]
just a small update, it looks like every image is assigned to a new skyregion.

**Comment by ycendes** [comment]
There is also some discussion on this going on here: https://github.com/transientskp/aartfaac-imaging-pipeline/issues/18

**Comment by ycendes** [comment]
I did a few tests where I took some AARTFAAC images through pyse.py first, with the same parameters as when it's run through TraP, to see whether the sourcefinder behaves correctly both "before" and "after" the jump in sources.  In both cases, the sourcefinder would find the same number of sources (and within the errors in position!), rather than hundreds in one and then a handful in the next. (So in this case, if there were 4 sources in one image but hundreds in the other, there were now 4 sources found in both images, and they were the same sources listed.) This is consistent over other "jumps" in the data too.

I also ran a very small sample set of two images through the TraP after this to see what would happen with the same result, ie not hundreds in one and 4 in the other but just 4 sources in both.  I have also done extensive checking of coordinates and source positions in many of the AARTFAAC images (detailed more in the other issue), and haven't found anything unusual in the images.

The conclusion here is I can confirm that so far it doesn't appear there's anything intrinsically wrong with the AARTFAAC images themselves, but rather an issue with how TraP is handling the images.
